### DISCUSSION TITLE
Test Aliro analysis with AutoGen

### DISCUSSION THREAD

**Discussion by jay-m-dev** [EpistasisLab/Aliro#630] [Ideas]
Submit a prompt to AutoGen that contains Aliro experiment metadata for analysis. We're trying to see if multiple agents can have a conversation regarding Aliro results.
We may choose to create our own project for this, using ExecGPT as the base, if it's not too time consuming to build. Otherwise we can just leverage MS AutoGen and use it as is.

One point to consider down the road will be the issue of privacy. Using AutoGen may require the use of third-parties (Microsoft, OpenAI). While having our own build would let us use our own local models.

**Comment by jay-m-dev** [comment]
I've looked into the [autogen project](https://github.com/microsoft/autogen) source code and it seems we should be able to leverage this project. Either within Aliro itself or as as standalone service that can interact with Aliro.
For the robotics program (Caire3PO) we may be able to add autogen (or a fork of autogen) as a backend service in the currently suggested [Caire3PO project architecture](https://github.com/EpistasisLab/Caire3PO/discussions/7)

**Reply by miguelehernandez** [reply]
Thanks Jay, yes, I mentioned I would prefer expanding the ExecGPT and try this with it. We also talked about having Aliro communicate with this form of AutoGen for Model and data analysis.

**Reply by miguelehernandez** [reply]
Jun is working on a quick demo for Jason. Hopefully we could see it today.

**Reply by jay-m-dev** [reply]
Agreed. It's all doable. Essentially we need to create the agents, much like AutoGen does and have them have a conversation. Then, we'd have to add the audio (text-to-speech, speech-to-text) components. We would need to put together some requirements and design this carefully if the project is approved. 
This project can be built on top of the current [Talking Robot](https://github.com/EpistasisLab/Caire3PO/discussions/5) project we're discussing in the Caire3PO project. So we could say the Talking Robot is **phase 1**, and this would be **phase 2**

**Comment by jay-m-dev** [comment]
Jason wants to hold off on this until next year (2024)
